app/dao/jobs_dao.py

```diff
@@ -148,7 +149,7 @@ def dao_get_jobs_older_than_data_retention(notification_types, limit=None):
             .order_by(desc(Job.created_at))
         )
         if limit:
-            query = query.limit(limit)
+            query = query.limit(limit - len(jobs))
```

Comment:
This was the fix: make sure you never add too many jobs to the `jobs` list (ie there should be at most `limit` jobs there)